receiver/awsfirehosereceiver/receiver.go

```diff
@@ -296,12 +296,11 @@ func loadEncodingExtension[T any](host component.Host, encoding, signalType stri
 	return unmarshaler, nil
 }
 
-// encodingToComponentID converts an encoding string to a component ID using the given encoding as type.
+// encodingToComponentID attempts to parse the encoding string as a component ID.
 func encodingToComponentID(encoding string) (*component.ID, error) {
-	componentType, err := component.NewType(encoding)
-	if err != nil {
+	var id component.ID
```

Comment:
What if the `component.ID` is used in the configuration of the receiver instead?

Reply:
I think that's a reasonable idea, but I'd like to leave it for a followup - I think it would just be cosmetic, and it's a lot more involved.